spec/controllers/labels_controller_spec.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe LabelsController do
+  render_views
+
+  let(:user) { create(:user) }
+  let(:label) { create(:label) }
+
+  describe 'GET #search' do
+    context 'without being signed in' do
+      it 'redirects to other page' do
+        get :search, params: {search: label.name, page: 1}
+        expect(response.body).to match(%r{^<html><body>You are being.*redirected.*</body></html>$})
+      end
+    end
+
+    context 'with valid params' do
+      subject(:request) { get :search, params: {search: label.name, page: 1} }
```

Comment:
If you move this subject to the above `describe` context, you should be able to reuse it in the `without being signed in` context. It should work there as well, I assume 🤔

Reply:
Yes, that works.